internal/app/utils.go

```diff
@@ -139,15 +138,55 @@ func toYAML(file string, s *state) {
 	newFile.Close()
 }
 
-// substituteVarsInValuesFiles loops through the values/secrets files and substitutes variables into them.
-func substituteVarsInValuesFiles(s *state) {
+// substituteVarsInStaticFiles loops through the values/secrets files and substitutes variables into them.
+func substituteVarsInStaticFiles(s *state) {
 	for _, v := range s.Apps {
 		if v.ValuesFile != "" {
 			v.ValuesFile = substituteVarsInYaml(v.ValuesFile)
 		}
 		if v.SecretsFile != "" {
 			v.SecretsFile = substituteVarsInYaml(v.SecretsFile)
 		}
+		if (hooks{}) != v.Hooks {
+			if v.Hooks.PreInstall != "" {
```

Comment:
maybe if we convert the hooks structure into a map here we could make this code less verbose.